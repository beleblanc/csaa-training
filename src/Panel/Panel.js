import React from 'react';
import { useWizard } from '../Wizard/Wizard';

const Panel = ({children, pageNumber}) => {
    const {currentPageNumber, isLastPage} = useWizard();

        return (currentPageNumber === pageNumber) ? (<div className="wizard-panel">
            {children}
            <div className="notification-center">
                You are on page number: {currentPageNumber}
            </div>
        </div>) : null;
}

export default Panel;

 