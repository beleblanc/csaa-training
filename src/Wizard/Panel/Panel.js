import React from "react";
import { useWizardContext } from "../Wizard.hooks";

const Panel = ({ children, pageNumber }) => {
  const { currentPageNumber } = useWizardContext();

  return currentPageNumber === pageNumber ? (
    <div className="wizard-panel">
      {children}
      <div className="notification-center">
        You are on page numbers: {currentPageNumber}
      </div>
    </div>
  ) : null;
};

export default Panel;
