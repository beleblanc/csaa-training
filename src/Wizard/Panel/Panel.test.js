import React from "react";
import { screen, render } from "@testing-library/react";
import Panel from "./Panel";

jest.mock("../Wizard.hooks", () => {
  return {
    useWizardContext: () => ({ currentPageNumber: 4 }),
  };
});

describe("Panel", () => {
  it("should render when it has the right page number", () => {
    const { container } = render(
      <Panel pageNumber={4}>
        <div>This is some text</div>
      </Panel>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText("This is some text")).toBeInTheDocument();
  });

  it("should not render the panel when the is not in the current page", () => {
    render(
      <Panel pageNumber={3}>
        <div>This is some text</div>
      </Panel>
    );

    expect(screen.queryByText("This is some text")).not.toBeInTheDocument();
  });
});
