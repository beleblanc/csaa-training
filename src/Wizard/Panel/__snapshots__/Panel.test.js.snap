// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should render when it has the right page number 1`] = `
<div>
  <div
    class="wizard-panel"
  >
    <div>
      This is some text
    </div>
    <div
      class="notification-center"
    >
      You are on page numbers: 
      4
    </div>
  </div>
</div>
`;
